### PR TITLE
Support observer without any data calibration

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24924 Serialization for nn.quantized.functional modules
* **#24923 Support observer without any data calibration**
* #24804 Update mapping dictionary to support functionalmodules and pooling operations
* #24803 Handle empty qconfig for functional Modules
* #24789 Work around for bias quantization for conv and linear operators

Replace exception with warning for un initialized min/max values to support creation of quantized models without observers.

Differential Revision: [D16923660](https://our.internmc.facebook.com/intern/diff/D16923660/)

### CODE REVIEW

**Comment by jerryzh168** [comment]
I think you might need to rebase: https://github.com/pytorch/pytorch/commit/049284e14d150f949768e87a77d443e104452f59

**Comment by jerryzh168** [comment]
but why do we need this?

**Comment by raghuramank100** [comment]
> but why do we need this?

This is to support quantize with no eval function. In this case, the observer should still return a valid scale/zero-point in calcqparams without needing any data.

File: torch/quantization/observer.py [hunk 1]
```diff
@@ -52,20 +53,19 @@ def calculate_qparams(self):
...
         max_val = self.max_val
         if max_val is None or min_val is None:
-            raise Exception('must run observer before calling calculate_qparams!')
-        max_val, min_val = float(max_val), float(min_val)
-        min_val = min(0.0, min_val)
-        max_val = max(0.0, max_val)
-        if max_val == min_val:
-            scale = 1.0
-            zero_point = 0
+            warnings.warn("must run observer before calling calculate_qparams")
```
Comment (jerryzh168) [inline torch/quantization/observer.py]: is this scriptable?

**Review by jerryzh168 — APPROVED** [review]
please make sure the observer is still scriptable

**Pull request closed without merging**